actor/src/main/java/org/apache/pekko/util/Unsafe.java

```diff
@@ -15,47 +15,37 @@
 
 import org.apache.pekko.annotation.InternalApi;
 
-import java.lang.reflect.Field;
+import java.lang.invoke.MethodHandles;
+import java.lang.invoke.VarHandle;
 import java.nio.charset.StandardCharsets;
 
 /** INTERNAL API */
 @InternalApi
 public final class Unsafe {
-  private static final sun.misc.Unsafe instance;
-
-  private static final long stringValueFieldOffset;
+  private static final VarHandle stringValueFieldHandle;
   private static final int copyUSAsciiStrToBytesAlgorithm;
 
   static {
     try {
-      sun.misc.Unsafe found = null;
-      for (Field field : sun.misc.Unsafe.class.getDeclaredFields()) {
-        if (field.getType() == sun.misc.Unsafe.class) {
-          field.setAccessible(true);
-          found = (sun.misc.Unsafe) field.get(null);
-          break;
-        }
-      }
-      if (found == null) throw new IllegalStateException("Can't find instance of sun.misc.Unsafe");
-      else instance = found;
-
-      long fo;
+      VarHandle handle;
       try {
-        fo = instance.objectFieldOffset(String.class.getDeclaredField("value"));
-      } catch (NoSuchFieldException nsfe) {
-        // The platform's implementation of String doesn't have a 'value' field, so we have to use
-        // algorithm 0
-        fo = -1;
+        MethodHandles.Lookup lookup =
+            MethodHandles.privateLookupIn(String.class, MethodHandles.lookup());
+        handle = lookup.findVarHandle(String.class, "value", byte[].class);
+      } catch (NoSuchFieldException | IllegalAccessException e) {
+        // The platform's implementation of String doesn't have a 'value' field
+        // or the field is inaccessible, so we have to use algorithm 0.
+        // You need `--add-opens=java.base/java.lang=ALL-UNNAMED` or similar to access it.
+        handle = null;
```

Comment:
should we print this to error message like kyo?

Reply:
* this exact same issue exists in the old code
* sun.misc.Unsafe requires exactly the same level of access granted by --add-opens
* I'm not against logging something though
* the code does not fail here though if insufficient access, it just works with standard algorithm instead (ie String getBytes public API)

Reply:
@He-Pin is system.err.println ok for the new logging?

Reply:
I'm fine, but in kyo, they log it to logger, or just to the exception error msg